src/test/java/com/amazon/ion/Ion_1_1_RoundTripTest.kt

```diff
@@ -0,0 +1,356 @@
+        symbol ?: return false
+        if (symbol.sid == 2) return true
+        symbol.text ?: return false
+        return ION_VERSION_MARKER_REGEX.matches(symbol.assumeText())
+    }
+
+    companion object {
+        @JvmStatic
+        protected val ION = IonSystemBuilder.standard().build() as _Private_IonSystem
+        private val ION_VERSION_MARKER_REGEX = Regex("^\\\$ion_[0-9]+_[0-9]+$")
+
+        private fun ionText(text: String): Array<Any> = arrayOf(text, text.encodeToByteArray())
+        private fun ionBinary(name: String, bytes: String): Array<Any> = arrayOf(name, hexStringToByteArray(bytes))
+
+        // Arguments here are an array containing a String for the test case name, and a ByteArray of the test data.
+        @JvmStatic
+        fun testData() = listOf(
+            ionText("\$ion_1_1 true \$ion_1_0 true \$ion_1_1 true"),
+            ionBinary("Binary IVMs", "E0 01 01 EA 6F E0 01 00 EA 10 E0 01 01 EA 6F"),
+            ionBinary("{a:{$4:b}}", "E0 01 01 EA FD 0F 01 FF 61 D3 09 A1 62"),
+            ionText("""a::a::c::a::0 a::a::0"""),
+            ionText("""a::a::c::a::0 a::0"""),
+            ionText("""foo::bar::baz::false foo::0"""),
+            ionText("""a::b::c::0 d::0"""),
+            ionText("""a::0 b::c::d::0"""),
+            ionText("""a::b::c::d::0 a::b::c::0"""),
+            ionText("""a::b::c::d::0 a::0 a::0"""),
+            ionText("""abc"""),
+            ionText(
+                """
+                    ${'$'}ion_1_0
+                    ${'$'}ion_symbol_table::{
+                      symbols:[ "a", "b", "c", "d", "e" ]
+                    }
+                    $10 $11 $12 $13 $14
+                    ${'$'}ion_1_0
+                    ${'$'}ion_symbol_table::{
+                      symbols:[ "rock", "paper", "scissors", "lizard", "spock" ]
+                    }
+                    $10 $11 $12 $13 $14
+                """.trimIndent()
+            ),
+        ) + files().flatMap { f ->
+            val ion = ION.loader.load(f)
+            // If there are embedded documents, flatten them into separate test cases.
+            if (ion.size == 1 && ion.first().hasTypeAnnotation("embedded_documents")) {
+                (ion.first() as IonContainer).mapIndexed { i, ionValue ->
+                    arrayOf<Any>("${f.path}[$i]", (ionValue as IonString).stringValue().toByteArray(Charsets.UTF_8))
+                }
+            } else {
+                listOf(arrayOf<Any>(f.path, f.readBytes()))
+            }
+        }
+
+        @JvmStatic
+        fun files() = testdataFiles(
+            And(GLOBAL_SKIP_LIST, LOCAL_SKIP_LIST),
+            GOOD_IONTESTS_FILES
+        )
+
+        @JvmField
+        val LOCAL_SKIP_LIST = setOf(
+            // Has an unknown, imported symbol
+            "symbolTablesUnknownText.ion",
+            // Skipped because there are no user values in these, and IonReaderNonContinuableSystem will throw an exception.
```

Comment:
🗺️ These files can reproduce another one of the issues mentioned in the PR summary.